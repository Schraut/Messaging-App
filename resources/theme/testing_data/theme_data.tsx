import { Theme } from "../theme";

import COLORS from "../colors";
import FONTS from "../fonts";
import { spacing } from "../spacing";

export const THEME_DATA: Theme = {
  colors: COLORS["light"],
  font: FONTS,
  spacing: spacing,
};
