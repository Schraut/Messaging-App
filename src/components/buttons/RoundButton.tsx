import { StyleSheet, TouchableOpacity } from "react-native";
import { Theme, useTheme } from "../../../resources/theme/theme";

interface RoundButtonProps {
  onPress: () => void;
  color?: string;
  icon?: any;
}

const RoundButton = ({ onPress, icon, color }: RoundButtonProps) => {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  return (
    <TouchableOpacity
      style={[
        styles.button,
        { backgroundColor: color || theme.colors.staticGrey },
      ]}
      onPress={onPress}
    >
      {icon}
    </TouchableOpacity>
  );
};

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    button: {
      flexDirection: "row",
      width: theme.spacing(10),
      height: theme.spacing(10),
      borderRadius: theme.spacing(10),
      justifyContent: "center",
      alignItems: "center",
    },
  });

export default RoundButton;
