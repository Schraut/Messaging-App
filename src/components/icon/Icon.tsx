import { View, StyleSheet } from "react-native";
import { Theme, useTheme } from "../../../resources/theme/theme";
import {
  MaterialIcons,
  Ionicons,
  FontAwesome5,
  FontAwesome,
  Entypo,
  Octicons,
  Feather,
  AntDesign,
} from "@expo/vector-icons";

interface IconProps {
  iconName: string;
  iconType: string;
  size: number;
  color?: string;
}

const Icon = ({ iconName, iconType, size, color }: IconProps) => {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  // Map the iconType to the correct icon component
  const getIconComponent = (type: string) => {
    switch (type) {
      case "MaterialIcons":
        return MaterialIcons;
      case "Ionicons":
        return Ionicons;
      case "FontAwesome5":
        return FontAwesome5;
      case "FontAwesome":
        return FontAwesome;
      case "Entypo":
        return Entypo;
      case "Octicons":
        return Octicons;
      case "Feather":
        return Feather;
      case "AntDesign":
        return AntDesign;
      default:
        return MaterialIcons; // Default to MaterialIcons if type is not recognized
    }
  };

  // Get the appropriate Icon component based on the iconType
  const IconComponent = getIconComponent(iconType);

  return (
    <View>
      <IconComponent
        name={iconName}
        size={size}
        color={color || theme.colors.staticBlack}
      />
    </View>
  );
};

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.inverseBackground,
      alignItems: "center",
      justifyContent: "center",
    },
  });

export default Icon;
