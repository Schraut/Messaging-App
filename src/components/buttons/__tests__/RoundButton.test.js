import React from "react";
import renderer from "react-test-renderer";
import ThemeContent from "../../../../resources/theme/theme";
import { THEME_DATA } from "../../../../resources/theme/testing_data/theme_data";
import Icon from "../../icon/Icon";
import RoundButton from "../RoundButton";

const theme = THEME_DATA;

describe("Default Round Button", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <RoundButton
            onPress={() => console.log("Pressed")}
            color={theme.colors.staticGrey}
            icon={
              <Icon
                iconName="add"
                iconType="MaterialIcons"
                size={24}
                color={theme.colors.staticBlack}
              />
            }
          />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <RoundButton
            onPress={() => console.log("Pressed")}
            color={theme.colors.staticGrey}
            icon={
              <Icon
                iconName="add"
                iconType="MaterialIcons"
                size={24}
                color={theme.colors.staticBlack}
              />
            }
          />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
