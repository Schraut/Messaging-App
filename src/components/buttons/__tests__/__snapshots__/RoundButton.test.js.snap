// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Round Button renders in dark mode 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#7E7E7E",
      "borderRadius": 40,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
>
  <View>
    <View />
  </View>
</View>
`;

exports[`Default Round Button renders in light mode 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#7E7E7E",
      "borderRadius": 40,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "width": 40,
    }
  }
>
  <View>
    <View />
  </View>
</View>
`;
